### PR TITLE
Change jsp 2.3 2.2 3.0 facade bundles

### LINKED ISSUE #12417: Fix java.lang.IllegalStateException: jstl facade bundle can not be located
Build breaks have encountered the following error occasionally (see defect 274796) : 

`SRVE0280E: Extension processor failed to initialize in factory [com.ibm.ws.jsp.webcontainerext.JSPExtensionFactory@12cc15ad]: java.lang.IllegalStateException: jstl facade bundle can not be located from its capability`

Problem as described by Tom Watson: 

It appears this issue could be avoided if the com.ibm.ws.jsp.jstl.facade bundle could be uninstalled after the com.ibm.ws.jsp bundle.

The issue is the com.ibm.ws.jsp bundle registers a com.ibm.wsspi.webcontainer.extension.ExtensionFactory with com.ibm.ws.jsp.webcontainerext.JSPExtensionFactory but the JSPExtensionFactory tries to find the com.ibm.ws.jsp.jstl.facade bundle in its implementation of createJspClassloaderContext.  This leaves a window of time when the JSPExtensionFactory is still registered and used by the web container after the com.ibm.ws.jsp.jstl.facade has been uninstalled.  If the com.ibm.ws.jsp was first stopped before uninstalling com.ibm.ws.jsp.jstl.facade then the JSPExtensionFactory would be unregistered and no longer in use by the web container.  At that point it should be safe to uninstall the com.ibm.ws.jsp.jstl.facade bundle.  To do this we should simply be able to give the com.ibm.ws.jsp.jstl.facade bundle a start phase of CONTAINER_EARLY

The following should be set for the facade bundle for all the jsp feature versions.

com.ibm.ws.jsp.jstl.facade; start-phase:=CONTAINER_EARLY

The default start-phase is CONTAINER, having CONTAINER_EARLY will cause the facade to start before the jsp bundle AND cause it to be stopped after the jsp bundle is stopped on shutdown or when the jsp feature is removed dynamically while the server is running.

### PR DESCRIPTION
**Pull request by DaBusyB**
fixes #12417 Changed jsp 2.3, 2.2, 3.0 jstl facade start phase to container early.

### CODE REVIEW

**Comment by DaBusyB** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+arguments=
+auto.sync=false
+build.scans.enabled=false
+connection.gradle.distribution=GRADLE_DISTRIBUTION(VERSION(6.3))
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs:4]: I don't think you want to check this file in.

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs:4]: @jhanders34 do you think this eclipse file should be in the .gitignore?

File: dev/com.ibm.websphere.appserver.features/.project [hunk 2]
```diff
@@ -19,5 +24,6 @@
 	<natures>
 		<nature>org.eclipse.jdt.core.javanature</nature>
 		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/.project:27]: Please don't change these.

File: dev/com.ibm.websphere.appserver.features/.classpath [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="src" output="bin/main" path="src">
+		<attributes>
+			<attribute name="gradle_scope" value="main"/>
+			<attribute name="gradle_used_by_scope" value="main,test"/>
+		</attributes>
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/.classpath:17]: Please don't change this

File: dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs (hunk 1 above)
Comment (jhanders34) [reply dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs:4]: In general we do not use the buildship eclipse nature.  You will not have a good experience using the buildship gradle nature in eclipse for OpenLiberty.  It is best to import your projects as existing projects and not as gradle projects using the buildship plugin.  Were you seeing some benefits using the gradle buildship plugin?

Comment (DaBusyB) [reply dev/com.ibm.websphere.appserver.features/.settings/org.eclipse.buildship.core.prefs:4]: Thanks tevans78 and jhanders24. I'm Volod will be working with me today to see where I went wrong and review the delivery process.

**Comment by DaBusyB** [comment]
#build

**Comment by DaBusyB** [comment]
#run-libby-bot

**Comment by pnicolucci** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature [hunk 4]
```diff
@@ -0,0 +1,48 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.jsp-3.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature:2]: I think this should be `io.openliberty.jsp-3.0`

File: dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+ org.apache.taglibs.standard.tag.rt.xml;  type="internal", \
+ org.apache.el;  type="internal", \
+ org.apache.el.lang;  type="internal", \
+ org.apache.jasper.el;  type="internal"
+IBM-ShortName: jsp-3.0
+IBM-SPI-Package: com.ibm.wsspi.jsp.taglib.config
+Subsystem-Name: Jakarta Server Pages 3.0
+-features=io.openliberty.jakarta.jsp-3.0, \
+ com.ibm.websphere.appserver.servlet-5.0; \
+ com.ibm.websphere.appserver.el-4.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature:37]: this should be `io.openliberty.el-4.0`

File: dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+-bundles=com.ibm.ws.org.eclipse.jdt.core.3.10.2.v20160712-0000, \
+ io.openliberty.jakarta.jstl.2.0; location:="dev/api/spec/,lib/"; mavenCoordinates="jakarta.servlet:jstl:2.0", \
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/public/jsp-3.0/com.ibm.websphere.appserver.jsp-3.0.feature:39]: maven co-ords look wrong, I think it should be `jakarta.servlet.jsp.jstl:jakarta.servlet.jsp.jstl-api:2.0.0`

**Comment by DaBusyB** [comment]
#build

**Comment by DaBusyB** [comment]
#run-libby-bot

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Pull request closed without merging**